app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -276,29 +292,44 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
                     getColorForStatus(txDetails.txStatus)
                 )
             }
-            is TransactionInfo.SettingsChange -> {
+            is TransactionInfoViewData.SettingsChange -> {
                 val viewStub = binding.stubSettingsChange
                 if (viewStub.parent != null) {
                     contentBinding = TxDetailsSettingsChangeBinding.bind(viewStub.inflate())
                 }
                 val txDetailsSettingsChangeBinding = contentBinding as TxDetailsSettingsChangeBinding
 
-                txDetailsSettingsChangeBinding.txAction.setActionInfoItems(txInfo.txActionInfoItems())
+                txDetailsSettingsChangeBinding.txAction.setActionInfoItems(txInfo.txActionInfoItems(requireContext().resources))
                 txDetailsSettingsChangeBinding.txStatus.setStatus(
                     TxType.MODIFY_SETTINGS.titleRes,
                     TxType.MODIFY_SETTINGS.iconRes,
                     getStringResForStatus(txDetails.txStatus, awaitingYourConfirmation),
                     getColorForStatus(txDetails.txStatus)
                 )
             }
-            is TransactionInfo.Custom -> {
+            is TransactionInfoViewData.Custom -> {
                 val viewStub = binding.stubCustom
                 if (viewStub.parent != null) {
                     contentBinding = TxDetailsCustomBinding.bind(viewStub.inflate())
                 }
                 val txDetailsCustomBinding = contentBinding as TxDetailsCustomBinding
-
-                txDetailsCustomBinding.txAction.setActionInfo(true, txInfo.formattedAmount(balanceFormatter), txInfo.logoUri()!!, txInfo.to)
+                val addressUri = when (val toInfo = txInfo.toInfo) {
+                    is AddressInfoData.Remote -> toInfo.addressLogoUri
+                    else -> null
+                }
+                val addressName = when (val toInfo = txInfo.toInfo) {
```

Comment:
see comment above. You should access name directly. All logic/checking should be done at the view model level

Reply:
Done in 4c8e06ae9940a95c1a08494abba7ec5f73d176e5